mat64/eigen.go

```diff
@@ -798,22 +800,23 @@ func hqr2(d, e []float64, hess, v *Dense, epsilon float64) {
 	}
 }
 
-// D returns the block diagonal eigenvalue matrix from the real and imaginary
-// components d and e.
-func (f EigenFactors) D() *Dense {
-	d, e := f.d, f.e
-	var n int
-	if n = len(d); n != len(e) {
-		panic(ErrSquare)
-	}
-	dm := NewDense(n, n, nil)
-	for i := 0; i < n; i++ {
-		dm.set(i, i, d[i])
-		if e[i] > 0 {
-			dm.set(i, i+1, e[i])
-		} else if e[i] < 0 {
-			dm.set(i, i-1, e[i])
+type EigenValues []complex128
+
+// D returns the eigenvalue matrix from the real and imaginary components
+// of the eigenvalues.
+//
+// If the decomposed matrix is not symmetric, then D returns a block diagonal
+// matrix with the real eigenvalues in 1-by-1 blocks and any complex eigenvalues,
+// lambda + i*mu, in 2-by-2 blocks, [lambda, mu; -mu, lambda].
+func (v EigenValues) D() *Dense {
```

Comment:
I'm still looking at where, but the Hessenberg matrix appears to be improperly zeroed

Matlab:
````
hess(a)

ans =

    1.0000   -9.0894   -1.5435
  -17.4929    8.0588    1.7647
         0    5.7647   -6.0588
````
Go:
````
[1 -9.08942501255297 -1.5434872662825807 
-17.492855684535904 8.05882352941176 1.7647058823529393 
15 5.76470588235294 -6.058823529411765]
````
Instead of a 0, the go version has a 15